Network/Network.cpp

```diff
@@ -625,154 +654,277 @@ namespace WPEFramework
+
+            if (getIPSettingsInternal(internal,InternalResponse))
+            {
+                response["interface"] = InternalResponse["interface"];
+                response["ipversion"] = InternalResponse["ipversion"];
+                response["autoconfig"] = InternalResponse["autoconfig"];
+                response["ipaddr"] = InternalResponse["ipaddr"];
+                response["netmask"] = InternalResponse["netmask"];
+                response["gateway"] = InternalResponse["gateway"];
+                response["primarydns"] = InternalResponse["primarydns"];
+		response["secondarydns"] = InternalResponse["secondarydns"];
+                result = true;
+            }
+            returnResponse(result)
+        }
+
+	uint32_t Network::getIPSettings2(const JsonObject& parameters, JsonObject& response)
+        {
+            JsonObject internal;
+	    JsonObject InternalResponse;
+            NetworkManager_GetIPSettings_ErrorCode_t errCode = NETWORK_IPADDRESS_ACQUIRED;
+	    bool result = false;
+            string interface;
+            string ipversion;
+            internal ["interface"] = interface;
+            internal ["ipversion"] = ipversion;
+			
+	    if (getIPSettingsInternal(internal,InternalResponse))
+            {
+                /*If the device was configured to use autoconfig IP but device does not have valid IP yet Could be the Router does not have DHCP Server running or
+                 * the device is in the process of acquiring it from the router),it must return only the autoconfig and the interface name. */
+		 if ( InternalResponse["ipaddr"].Content() == Core::JSON::Variant::type::EMPTY || errCode == NETWORK_IPADDRESS_NOTFOUND )
+		 {
+		    response["interface"] = InternalResponse["interface"];
+                    response["autoconfig"] = InternalResponse["autoconfig"];
+                    result = true;
+                    returnResponse(result)
+		 }
+		 if (errCode == NETWORK_IPADDRESS_ACQUIRED)
+		 {
+     		     response["interface"] = InternalResponse["interface"];
+                     response["ipversion"] = InternalResponse["ipversion"];
+                     response["autoconfig"] = InternalResponse["autoconfig"];
+	             response["ipaddress"] = InternalResponse["ipaddr"];
+                     response["netmask"] = InternalResponse["netmask"];
+                     response["gateway"] = InternalResponse["gateway"];
+                     response["primarydns"] = InternalResponse["primarydns"];
+                     //If the secondaryDNS was not set , it shouldn't return secondaryDNS in response.
+		     if ( InternalResponse["secondarydns"].Content() != Core::JSON::Variant::type::EMPTY )  
+		         response["secondarydns"] = InternalResponse["secondarydns"];
+                     result = true;
+                }
+	    }
+	    returnResponse(result)
+        }
+	uint32_t Network::getIPSettingsInternal(const JsonObject& parameters, JsonObject InternalResponse)
```

Comment:
keep the input variable name as _response_ instead of _InternalResponse_.
This will help the return code and other existing macros